src/clusterfuzz/_internal/bot/tasks/utasks/minimize_task.py

```diff
@@ -586,30 +574,69 @@ def utask_main(uworker_input):
     testcase.put()
 
     if check_deadline_exceeded_and_store_partial_minimized_testcase(
-        deadline, uworker_input.testcase_id, uworker_input.job_type,
-        input_directory, file_list, data, testcase_file_path):
-      return
+        deadline, testcase, uworker_input.job_type, input_directory, file_list,
+        data, testcase_file_path):
+      return None
 
   command = testcase_manager.get_command_line_for_application(
       testcase_file_path, app_args=app_arguments, needs_http=testcase.http_flag)
   last_crash_result = test_runner.last_failing_result
 
-  store_minimized_testcase(testcase, input_directory, file_list, data,
-                           testcase_file_path)
-  finalize_testcase(
-      uworker_input.testcase_id,
-      command,
-      last_crash_result,
+  return uworker_io.UworkerOutput(
+      testcase=testcase,
+      input_directory=input_directory,
+      file_list=file_list,
+      data=data,
+      testcase_file_path=testcase_file_path,
+      command=command,
+      last_crash_result=last_crash_result,
       flaky_stack=flaky_stack)
 
 
-def finalize_testcase(testcase_id,
-                      command,
-                      last_crash_result,
-                      flaky_stack=False):
+HANDLED_ERRORS = [
+    uworker_msg_pb2.ErrorType.MINIMIZE_SETUP,
+    uworker_msg_pb2.ErrorType.TESTCASE_SETUP,
+    uworker_msg_pb2.ErrorType.TESTCASE_SETUP_INVALID_FUZZER,
+    uworker_msg_pb2.ErrorType.UNHANDLED
+]
+
+
+def utask_postprocess(output):
+  """Postprocess in a trusted bot."""
+  if output.error is not None:
+    uworker_handle_errors.handle(output, HANDLED_ERRORS)
+    return
+
+  store_minimized_testcase(output.testcase, output.input_directory,
+                           output.file_list, output.data,
+                           output.testcase_file_path)
+  finalize_testcase(
+      output.testcase,
+      output.command,
+      output.last_crash_result,
+      flaky_stack=output.flaky_stack)
+
+
+def handle_minimize_setup_error(output):
+  """Handles errors occuring during setup."""
+
+  if output.original_job_name:
+    _skip_minimization(output.testcase,
+                       'Failed to setup build for overridden job.')
+  else:
+    # Only recreate task if this isn't an overriden job. It's possible that a
+    # revision exists for the original job, but doesn't exist for the
+    # overriden job.
+    tasks.add_task(
+        'minimize',
+        output.testcase.key.id(),
```

Comment:
Let's use output.uworker_input.testcase, it's actually trusted.

Reply:
output doesnt have a uworker_input field at the moment - should I also be passing that in?

Reply:
It's added automatically, don't pass it.

Reply:
It's added automatically during deserialization (to ensure it's actually the right data instead of only what uworker claims was the input), don't add it please.

Reply:
ah gotcha - done!